include/rtc.h

```diff
@@ -0,0 +1,97 @@
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef RTC_H_
+#define RTC_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct rtc_dev
+ * @brief Structure holding RTC descriptor.
+ */
+struct rtc_dev {
+	/** ID of the real time clock core. */
+	uint8_t id;
+	/** Device specific RTC parameters. */
+	void *extra;
+};
+
+/**
+ * @struct rtc_init_param
+ * @brief Structure holding RTC initialization parameters.
+ */
+struct rtc_init_param {
+	/** ID of the real time clock core. */
+	uint8_t id;
+	/** Device specific RTC parameters. */
+	void *extra;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/** Initialize the RTC peripheral. */
+int32_t rtc_init(struct rtc_dev **device, struct rtc_init_param *init_param);
+
+/** Free the resources allocated by rtc_init(). */
+int32_t rtc_remove(struct rtc_dev *dev);
+
+/** Start the real time clock. */
+int32_t rtc_start(struct rtc_dev *dev);
+
+/** Stop the real time clock. */
+int32_t rtc_stop(struct rtc_dev *dev);
+
+/** Get the current count for the real time clock. */
+int32_t rtc_get_cnt(struct rtc_dev *dev, uint32_t *tmr_cnt);
```

Comment:
I whould use a uint64_t tmr_cnt. This is because if we want to count the seconds since January 1, 1970, the uin32_t will overflow soon.

Reply:
The count register for the ADuCM3029 RTC is a 32 bit register. More accurately it is 2 16 bit registers that function together. It does contain another 15 bit register, but that can only be read as a fractional part to the current value. It is the duty of the application to provide time measurement using this function an interrupts.

Reply:
I know. I'am not strongly want this change, but I still think that doing this inside the driver will ease the application development